flowcraft/generator/components/typing.py

```diff
@@ -161,4 +161,101 @@ def __init__(self, **kwargs):
         ]
 
 
+class Seqsero2Reads(Process):
+    """SeqSero2 for reads process template interface
 
+    This process is set with:
+
+        - ``input_type``: fastq
+        - ``output_type``: None
+        - ``ptype``: typing
+    """
+
+    def __init__(self, **kwargs):
+
+        super().__init__(**kwargs)
+
+        self.input_type = "fastq"
+        self.output_type = None
+
+        self.directives = {
+            "seqsero2_reads": {
+                "cpus": 1,
+                "memory": "{ 1.GB * task.cpus * task.attempt }",
+                "container": "ummidock/seqsero2",
+                "version": "alpha-test-1",
+                "cache": "false",
+                "scratch": "true"
+            }
+        }
+
+
+class Seqsero2Assembly(Process):
+    """SeqSero2 for assembly process template interface
+
+    This process is set with:
+
+        - ``input_type``: fasta
+        - ``output_type``: None
+        - ``ptype``: typing
+    """
+
+    def __init__(self, **kwargs):
+
+        super().__init__(**kwargs)
+
+        self.input_type = "fasta"
+        self.output_type = None
+
+        self.directives = {
+            "seqsero2_assembly": {
+                "cpus": 1,
+                "memory": "{ 1.GB * task.cpus * task.attempt }",
+                "container": "ummidock/seqsero2",
+                "version": "alpha-test-1",
+                "cache": "false",
+                "scratch": "true"
+            }
+        }
+
+
+class StxSeqtypingReads(Process):
+    """ecoli_stx_subtyping.py for reads process template interface
+
+    This process is set with:
+
+        - ``input_type``: fastq
+        - ``output_type``: None
+        - ``ptype``: typing
+    """
+
+    def __init__(self, **kwargs):
+
+        super().__init__(**kwargs)
+
+        self.input_type = "fastq"
+        self.output_type = None
+
+        self.params = {
+            "stx2covered": {
+                "default": 'null',
+                "description": "Minimal percentage of sequence covered to consider "
+                               "extra stx2 subtypes (value between [0, 100]) (default: 100)."
+            },
+            "stx2identity": {
+                "default": 'null',
+                "description": "Minimal sequence identity to consider extra stx2 "
+                               "subtypes (value between [0, 100]) (default: 99.5)."
+            }
+        }
+
+        self.directives = {
+            "stx_seqtyping_reads": {
+                "cpus": 4,
+                "memory": "{ 1.GB * task.cpus * task.attempt }",
```

Comment:
same as the above comment

Reply:
This is one of those cases in which the memory used vary with the CPUs: 594.6 MB with 1 CPU, 879.3 MB with 2 CPUs and 1.5 GB with 4 CPUs. I'll leave this one.

Reply:
I see your point but I am concerned that this may cause issues in some personal computers. The user can always change it anyway in the config or even using the pipeline string, so it's not that big of a deal.

Reply:
Removed in SeqSero2 components and I added more information about this in the docs for the one that remains.